test/plugin.test.js

```diff
@@ -81,42 +81,113 @@ QUnit.test('exposes options', function(assert) {
                      'exposes publisherId');
 });
 
-QUnit.module('plugin guard functions');
+QUnit.module('plugin guard functions', {
+  beforeEach() {
+    this.options = {
+      keySystems: {
+        'org.w3.clearkey': {}
+      }
+    };
 
-QUnit.test('handleEncryptedEvent checks for required options', function(assert) {
-  const initData1 = new Uint8Array([1, 2, 3]).buffer;
-  const event = {
+    this.initData1 = new Uint8Array([1, 2, 3]).buffer;
+    this.initData2 = new Uint8Array([4, 5, 6]).buffer;
+
+    this.event1 = {
     // mock video target to prevent errors since it's a pain to mock out the continuation
     // of functionality on a successful pass through of the guards
-    target: {},
-    initData: initData1
-  };
-  let options = {};
+      target: {},
+      initData: this.initData1
+    };
+    this.event2 = {
+      target: {},
+      initData: this.initData2
+    };
+
+    this.origRequestMediaKeySystemAccess = navigator.requestMediaKeySystemAccess;
+
+    navigator.requestMediaKeySystemAccess = (keySystem, options) => {
+      return Promise.resolve({
+        keySystem: 'org.w3.clearkey',
+        createMediaKeys: () => {
+          return {
+            createSession: () => new videojs.EventTarget()
+          };
+        }
+      });
+    };
+  },
+  afterEach() {
+    navigator.requestMediaKeySystemAccess = this.origRequestMediaKeySystemAccess;
+  }
+});
+
+QUnit.test('handleEncryptedEvent checks for required options', function(assert) {
+  const done = assert.async();
   let sessions = [];
 
-  handleEncryptedEvent(event, options, sessions);
-  assert.equal(sessions.length, 0, 'did not create a session when no options');
+  handleEncryptedEvent(this.event1, {}, sessions).then(() => {
+    assert.equal(sessions.length, 0, 'did not create a session when no options');
+    done();
+  });
+});
 
-  options = { keySystems: {} };
-  handleEncryptedEvent(event, options, sessions);
-  assert.equal(sessions.length, 1, 'created a session when keySystems in options');
-  assert.equal(sessions[0].initData, initData1, 'captured initData in the session');
+QUnit.test('handleEncryptedEvent creates session', function(assert) {
+  const done = assert.async();
+  let sessions = [];
+
+  handleEncryptedEvent(this.event1, this.options, sessions).then(() => {}, () => {
```

Comment:
Is this testing a rejection?

Reply:
Also for below

Reply:
Going to add a comment

Reply:
Minor, but should we put this comment for each use?